addons/mrp/models/mrp_production.py

```diff
@@ -221,6 +224,56 @@ def _get_default_date_planned_finished(self):
     delivery_count = fields.Integer(string='Delivery Orders', compute='_compute_picking_ids')
     confirm_cancel = fields.Boolean(compute='_compute_confirm_cancel')
 
+    will_be_late = fields.Boolean(compute='_compute_will_be_late', default=False, store=True)
+    latest_previous_mrp_production_id = fields.Many2one('mrp.production', string='Previous MO', compute='_compute_latest_previous_mrp_production_id')
+
+    @api.depends('move_raw_ids.date_expected')
+    def _compute_date_planned_start(self):
+        for mrp_production in self:
+            if mrp_production.move_raw_ids:
+                mrp_production.date_planned_start = max(mrp_production.mapped('move_raw_ids.date_expected'))
+            else:
+                mrp_production.date_planned_start = fields.Datetime.now()
+
+    def _set_date_planned_start(self):
+        for mrp_production in self:
```

Comment:
split in a proper commit the removal of the onchange for the compute with an inverse please